src/antlr/terraformParser.g4

```diff
@@ -7,7 +7,7 @@ file
   ;
 
 dataDirective
-  : DATA providerType name AO (object)+ AF
+  : DATA providerType name AO object* AF
```

Comment:
why not for other terraform blocks (provider, module, terraform, variable) ?

Reply:
The `terraform validate` command throws an error when no `source` argument is provided to a module block,or no `value` argument is provided to an output block. However,  the other directives can indeed have an empty body. I changed it accordingly.